neqo-transport/src/frame.rs

```diff
@@ -268,45 +268,34 @@ impl Frame {
             overhead += Encoder::varint_len(offset);
         }
 
-        if overhead > space {
-            qdebug!(
-                "Frame::new_stream -> None; hdrlen {} > space {}",
-                overhead,
-                space
-            );
-            return None;
-        }
-
-        let (fin, fill) = if data.len() > (space - overhead) {
+        let (fin, fill) = match (data.len() + overhead).cmp(&space) {
             // More data than fits, fill the packet and negate |fin|.
-            (false, true)
-        } else if data.len() == (space - overhead) {
+            Ordering::Greater => (false, true),
             // Exact fit, fill the packet, keep |fin|.
-            (fin, true)
-        } else {
+            Ordering::Equal => (fin, true),
             // Too small, so include a length.
-            let data_len = min(space - overhead - 1, data.len());
-            overhead += Encoder::varint_len(u64::try_from(data_len).unwrap());
+            Ordering::Less => {
+                let data_len = min(space.saturating_sub(overhead + 1), data.len());
```

Comment:
Gotcha here: I avoid the need to check overhead vs. space twice by using saturating_sub here to avoid underflow.